hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -515,4 +507,9 @@ private void alertTokenBackingStoreOfNew(TokenRelationship newRel) {
 	public interface LegacyTreasuryRemover {
 		void removeKnownTreasuryForToken(final AccountID aId, final TokenID tId);
 	}
+
+	@FunctionalInterface
```

Comment:
**NIT**

You can check and try implementing a similar approach:
https://github.com/LimeChain/hedera-services/blob/60610a6212b97cd9b49cdd5a67df9dc7d83168da/hedera-node/src/main/java/com/hedera/services/store/schedule/ExtantSchedulesDelegator.java